examples/basics/index.js

```diff
@@ -0,0 +1,81 @@
+'use strict'
+
+const fs = require('fs')
+const os = require('os')
+const series = require('async/series')
+const IPFS = require('../../src/core') // replace this by line below
+// const IPFS = require('ipfs')
+
+/*
+ * Create a new IPFS instance, using default repo (fs) on default path (~/.ipfs)
+ */
+const node = new IPFS(os.tmpDir() + '/' + new Date().toString())
+
+const fileToAdd = {
+  path: 'hello.txt',
+  content: fs.createReadStream('./hello.txt')
+}
+
+let fileMultihash
+
+series([
+  /*
+   * Display version of js-ipfs
+   */
+  (cb) => {
+    node.version((err, version) => {
+      if (err) { return cb(err) }
+
+      console.log('IPFS Version:', version.version)
+      cb()
+    })
+  },
+  /*
+   * Initialize the repo for this node
+   */
+  (cb) => node.init({ emptyRepo: true, bits: 2048 }, cb),
+  /*
+   * Load the repo config into the IPFS node
+   */
+  (cb) => node.load(cb),
+  /*
+   * Take the node online (bitswap, network and so on)
+   */
+  (cb) => node.goOnline(cb),
+  /*
+   * Add a file to IPFS - Complete Files API on:
+   * https://github.com/ipfs/interface-ipfs-core/tree/master/API/files
+   */
+  (cb) => {
+    if (node.isOnline()) {
+      console.log('\nNode is now ready and online')
+    }
+
+    node.files.add(fileToAdd, (err, result) => {
+      if (err) { return cb(err) }
+
+      console.log('\nAdded file:')
+      console.log(result[0])
+      fileMultihash = result[0].hash
```

Comment:
Instead of doing mutation on the fileMultihash variable, you should be able to pass the value as a second parameter to the callback and then retrieve it in the next step, if I remember correctly.